earth_enterprise/src/fusion/autoingest/plugins.src/CombinedRP.src

```diff
@@ -178,8 +178,13 @@ CombinedRPAssetVersionImplD::UpdateChildren(void)
 
     // see if I need to khvr the source
     if (config.need_khvr_) {
-      source = MosaicFactory::ReuseOrMakeAndUpdateSubAsset
-               (assetRef, type, "mosaic",
+      source =
+        AssetFactory::ReuseOrMakeAndUpdateSubAsset
+        <MutableMosaicAssetVersionD, AssetVersion,
+         MosaicConfig, MosaicAssetVersionD>
```

Comment:
Just a general comment here: the template parameters `MutableMosaicAssetVersionD`, `MosaicConfig`, `MosaicAssetVersionD` always have to be used together. Mixing a MosaicConfig with a SourceAssetVersionD, for example, wouldn't work. Because of that, it would be nice if we could just specify one template parameter that covers all 3. For example, perhaps `MutableMosaicAssetVersionD::Impl::ConfigType` could be `MosaicConfig` and then you don't have to pass the Config parameter. This might not be worth the amount of work it would take, but it's worth considering.

For that matter, I think AssetVersion is always the second template parameter, so we should just hard code that.

In general it would be nice to have fewer template parameters. 4 is a lot. Again, it may not be possible to reduce them, but it would be nice.

Reply:
A reminder on this one: the AssetVersion template parameter seems unnecessary. It's never set to anything other than AssetVersion, so we could just hard-code it in the functions themselves. That said, it's not a big deal, so I'm OK with leaving it as is.